spur/ssh.py

```diff
@@ -80,7 +80,14 @@ def generate_run_command(self, command_args, store_pid,
             commands.append("echo $$")
 
         if cwd is not None:
-            commands.append("cd {0}".format(escape_sh(cwd)))
+            # 1. Change directory before to check if the command exists
+            # (using command || which) because it might be in this
+            # directory:
```

Comment:
In which case, there should probably be a test case for that behaviour. Also, no need to spell out exactly how we're going to check whether the command exists -- it should suffice simply to state that we change directory before checking the command exists, although hopefully that should be evident from the code (and if it's not, then it's worth thinking about whether the code be rejigged to make that more apparent).